### ISSUE TITLE
context loading failed: no go files to analyze

### ISSUE THREAD

**Issue by rmilejcz** [golangci/golangci-lint#825]
I have regularly run `golangci-lint run` from the root of my directory without issue for the last several weeks. Today, after successfully running `golangci-lint run`, I fixed all the reported issues and then ran the command in the same terminal but I got the output:

```
ERRO Running error: context loading failed: no go files to analyze
```
even though nothing seems to have changed. The project structure is basically 60 or so folders, each having a `main.go`, a `main_test.go` and two auto generated files which are ignored by `golangci-lint`. If I specify a package it lints without issue e.g.:

```
golangci-lint run ActivityLog
```
runs the linter on that package without any error.

<details><summary>Version of golangci-lint</summary>

```bash
$ golangci-lint --version
golangci-lint has version 1.21.0 built from 645e794 on 2019-10-15T18:15:04Z
```

</details>

<details><summary>Config file</summary>

```bash
$ cat .golangci.yml
run:
  concurrency: 4
  deadline: 2m
  issues-exit-code: 1
  tests: true

output:
  format: colored-line-number
  print-issued-lines: true
  print-linter-name: true

linter-settings:
  errcheck:
    check-type-assertions: true
    check-blank: true

  govet:
    check-shadowing: true
    settings:
      printf:
        funcs:
          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Infof
          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Warnf
          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Errorf
          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Fatalf
  golint:
    min-confidence: 0.8

  gofmt:
    simplify: true

  goimports:
    local-prefixes: kalos-core

  gocyclo:
    min-complexity: 20

  maligned:
    suggest-new: true

  goconst:
    min-len: 3
    min-occurrences: 3

  misspell:
    locale: US
    ignore-words:
      - supercalifragilisticexpialidocious

  lll:
    line-length: 120
    tab-width: 2

  unused:
    check-exported: false

  unparam:
    check-exported: false

  prealloc:
    simple: true
    range-loops: true
    for-loops: false

  gocritic:
    enabled-checks:
      - appendAssign
      - argOrder
      - assignOp
      - badCond
      - boolExprSimplify
      - builtinShadow
      - captLocal
      - caseOrder
      - commendOutCode
      - commentFormatting
      - commentedOutImport
      - defaultCaseOrder
      - deprecatedComment
      - docStub
      - dupArg
      - dupBranchBody
      - dupCase
      - dupSubExpr
      - elseif
      - emptyFallthrough
      - emptyStringTest
      - equalFold
      - exitAfterDefer
      - flagDeref
      - flagName
      - hexLiteral
      - hugeParam
      - ifElseChain
      - importShadow
      - indexAlloc
      - initClause
      - methodExprCall
      - nestingReduce
      - nilValReturn
      - octalLiteral
      - offBy1
      - paramTypeCombine
      - ptrToRefParam
      - rangeExprCopy
      - rangeValCopy
      - regexpMust
      - singleCaseSwitch
      - sloppyLen
      - sloppyReassign
      - stringXbytes
      - switchTrue
      - typeAssertChain
      - typeSwitchVar
      - typeUnparen
      - underef
      - unlabelStmt
      - unlambda
      - unnecessaryBlock
      - unslice
      - valSwap
      - weakCond
      - wrapperFunc
      - yodaStyleExpr

    disabled-checks:
      - codegenComment
      - unnamedResult

linters:
  enable:
    - bodyclose
    - megacheck
    - govet
    - errcheck
    - staticcheck
    - unused
    - gosimple
    - structcheck
    - varcheck
    - ineffassign
    - deadcode
    - typecheck
    - gosec
    - interfacer
    - unconvert
    - goconst
    - gocyclo
    - goimports
    - maligned
    - misspell
    - unparam
    - scopelint
    - gocritic
    - gochecknoinits
  disable:
    - dupl
    - lll
    - stylecheck
```

</details>

<details><summary>Go environment</summary>

```bash
$ go version && go env
go version go1.13.1 darwin/amd64
GO111MODULE="on"
GOARCH="amd64"
GOBIN="/Users/robertmilejczak/Documents/Projects/bin"
GOCACHE="/Users/robertmilejczak/Library/Caches/go-build"
GOENV="/Users/robertmilejczak/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/robertmilejczak/Documents/Projects"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.13.1/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.13.1/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/robertmilejczak/Documents/Projects/src/github.com/rmilejcz/kalos-core/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/kv/npdslxvd0ml09ntv_0rsmqcr0000gn/T/go-build078154239=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

<details><summary>Verbose output of running</summary>

```bash
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/robertmilejczak/Documents/Projects/src/github.com/rmilejcz/kalos-core/server /Users/robertmilejczak/Documents/Projects/src/github.com/rmilejcz/kalos-core /Users/robertmilejczak/Documents/Projects/src/github.com/rmilejcz /Users/robertmilejczak/Documents/Projects/src/github.com /Users/robertmilejczak/Documents/Projects/src /Users/robertmilejczak/Documents/Projects /Users/robertmilejczak/Documents /Users/robertmilejczak /Users /] 
INFO [config_reader] Used config file ../.golangci.yml 
INFO [lintersdb] Active 23 linters: [bodyclose deadcode errcheck gochecknoinits goconst gocritic gocyclo goimports gosec gosimple govet ineffassign interfacer maligned misspell scopelint staticcheck structcheck typecheck unconvert unparam unused varcheck] 
INFO [loader] Go packages loading at mode 575 (types_sizes|deps|imports|name|compiled_files|exports_file|files) took 1.436796726s 
ERRO Running error: context loading failed: no go files to analyze 
INFO Memory: 16 samples, avg is 68.9MB, max is 68.9MB 
INFO Execution took 1.451511334s
```

</details>

**Comment by rmilejcz** [comment]
Disregard, for those getting a similar error message make sure you can build your project. I was able to figure out the issue by running 

```
go test -v ./...
```

If possible, the error message `context loading failed: no go files to analyze` should be improved, especially when specifying verbose ouput.

**Comment by jirfag** [comment]
thank you, I think we need to fix error message

**Comment by alejandro-delgado-garcia** [comment]
Hi, I'm having the same issue even in latest version 1.21.0, any ideas what could be the issue? Thanks.

**Comment by rmilejcz** [comment]
The only advice I could give is to make sure your code is error free, run `go build` or `go test` to make sure that everything works as expected

**Comment by dackroyd** [comment]
This is tripping up our code formatting, as we've moved that to GolangCI Lint (running `gofmt` and `goimports`) rather than running those tools using locally installed versions. This is to ensure that a consistent toolset is used across all developers machines.

Goimports would normally clean up any import paths that don't exist (e.g. from code that has been removed), however when running in GolangCI Lint, this gives the `no go files to analyze` error instead

**Comment by alejandro-delgado-garcia** [comment]
We had to downgrade to 1.17 to make it to work and about goimports we had to disable it and use it standalone without golangci lint. Maybe we are doing something wrong but at least we found a way. I think the linter doesn't play nice with a multi go modules repository in our case, as we are using a mono repository with multiple go modules in it one per each service

**Comment by fdcds** [comment]
This issue has several duplicates: #855, #865, #870, #916.

**Comment by mr-tron** [comment]
For me it was connected to unupdated vendoring directory.

**Comment by sam-github** [comment]
In https://github.com/wtfutil/wtf/pull/895 I used `args: ./...` with golangci/golangci-lint-action@v0.2.0 to avoid this, though I don't need to when running on my own machine.

**Comment by jirfag** [comment]
Hi,
there at least two different cases:

1. no `go.mod` on the root, it's only in the subdirectory. I think it should be fixed by specifying a working directory because all built-in Go tooling doesn't work with such cases. I've made golangci/golangci-lint-action#15 for our GitHub action.
2. timeout occurred during `go/packages` loading. The bug is in `go/packages`. We need help in fixing it and making a pull request to [x/tools](https://github.com/golang/tools). The bug is in:
  * `pkgs, err := packages.Load(conf, args...)`
  * `func goListDriver(cfg *Config, patterns ...string) (*driverResponse, error) {`
  * `func (state *golistState) createDriverResponse(words ...string) (*driverResponse, error) {`
  * `func (state *golistState) invokeGo(verb string, args ...string) (*bytes.Buffer, error) {`
  *  `	stdout, stderr, _, err := gocmdRunner.RunRaw(cfg.Context, inv)` - here we ignore `firendlyErr` which contains a context cancellation error.
To reproduce it run in the repo of `golangci-lint`:

```bash
golangci-lint cache clean && go clean -modcache -cache -i
golangci-lint run -v --timeout=5s
```

**Comment by jirfag** [comment]
After merging #1154,
do you have examples of public repos where we can reproduce the problem with `no go files`?

**Comment by ryancurrah** [comment]
One way to reproduce this with golangci itself is to:

1) Set `modules-download-mode: readonly` in `golangci.yml`
2) Remove a few entries from `go.sum`
3) run golangci: 
```bash
╰─ ./golangci-lint run ./...
ERRO Running error: context loading failed: no go files to analyze
```

**Comment by ryancurrah** [comment]
Seems to be `packages.Load(conf, args...)` does not return an error even though the debug output with `export GL_DEBUG=loader` shows a debug line from `packages` of:

```bash
DEBU [loader] 191.382563ms for GOROOT=/Users/ryancurrah/.go/current GOPATH= GO111MODULE= GOPROXY= PWD=/Users/ryancurrah/git/golangci-lint go [go list -mod=readonly -e -json -compiled=true -test=true -export=true -deps=true -find=false -mod=readonly -- ./...]
```

https://github.com/golangci/golangci-lint/blob/master/pkg/lint/load.go#L200

**Comment by ryancurrah** [comment]
Ok after a bunch of debugging the `tools` package from golang I found the issue and created a Pull Request here: https://github.com/golang/tools/pull/231.

If this fix is accepted `packages.Load` will properly raise errors related to `go.sum` issues instead of returning a empty slice of packages. Then updating `golang.org/x/tools` dependancy will solve this particular issue. It may not solve all `no go files to analyze` issues because `go/packages/golist.go` function `invokeGo` selectively chooses which errors to raise. Meaning not all errors are caught and this is on purpose.

**Comment by ryancurrah** [comment]
Just another thought. Whenever someone is having this issue they should run:

```
go list -mod=readonly -e -json -compiled=true -test=true -export=true -deps=true -find=false -- ./...
```

In their project to see if a error occurs. If one does; another pull request to the tools package to catch that error may be required. 

If you don’t want to fix it, the command will at least tell you what’s really wrong. 

That command is the same one the tools package uses.

**Comment by rollulus** [comment]
We're observing the same, but what's not clear to me if others see this intermittently only or not (we do). We use the GitHub action, and rerunning sometimes fixes the issue. If there's more input I can provide, I'll be happy to do so.

```
2020-07-03T14:38:06.8296454Z Running [/home/runner/golangci-lint-1.27.0-linux-amd64/golangci-lint run --out-format=github-actions -E gofmt] in [/home/runner/work/xxx] ...
2020-07-03T14:39:06.9584039Z level=error msg="Running error: context loading failed: no go files to analyze"
2020-07-03T14:39:06.9645099Z level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
```

**Comment by wtask** [comment]
The same here with golangci-lint 1.28.3. After deletion `go.sum` and running `go mod tidy` the problem was resolved for me and golangci-lint successfully started. Note, in my case the `go.sum` is in the `.gitignore`

**Comment by nyetwurk** [comment]
`go clean -modcache` fixed this for me. An upstream dependency had a force push in its history, which was cached in the CI instance for some reason but not my local machine.

**Comment by nyetwurk** [comment]
For us an upstream repo had a force push in it that was cached in go.sum.

To fix (on local machine):
```bash
rm go.sum
go clean -modcache
go mod tidy
```
then commit new `go.sum`, push, and ci works again

**Comment by penglongli** [comment]
> For us an upstream repo had a force tag push in it that was cached in go.sum.
> 
> To fix (on local machine):
> 
> ```shell
> rm go.sum
> go clean -modcache
> go mod tidy
> ```
> 
> then commit new `go.sum`, push, and ci worked again
> 
> As @ryancurrah suggested
> 
> ```
> go list -mod=readonly -e -json -compiled=true -test=true -export=true -deps=true -find=false -- ./...
> ```
> 
> was very helpful and showed the error right away.

This fixed my problem.

**Comment by joemcmahon** [comment]
I got this error today on OS X because `~/Library/Caches/go-build` was owned by `root`. The `go list` command mentioned by others above also tracked this problem down.

**Comment by ybiquitous** [comment]
Hi there,

I notice that this problem occurs also with `golangci-lint` v1.37.1 (the latest version), while it never occurs with v1.36.0.

Here is the reproduction via Docker. ⬇️ 

Given `test.go` in the current directory as below:

```go
package main

var unused int

func main() {
	print()
}
```

Passed with v1.36.0:

```console
$ docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.36.0 golangci-lint run
test.go:3:5: `unused` is unused (deadcode)
var unused int
    ^

$ docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.36.0 golangci-lint version
golangci-lint has version v1.36.0 built from 6c25d068 on 2021-01-26T13:24:48Z

$ docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.36.0 go version
go version go1.15.7 linux/amd64
```

Failed with v1.37.1:

```console
$ docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.37.1 golangci-lint run
level=error msg="Running error: context loading failed: no go files to analyze"

$ docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.37.1 golangci-lint version
golangci-lint has version v1.37.1 built from b39dbcd6 on 2021-02-20T11:59:15Z

$ docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.37.1 go version
go version go1.16 linux/amd64
```

Maybe it seems due to Go 1.16... 🤔

**Comment by ldez** [comment]
I don't reproduce:

```console
$ golangci-lint version
golangci-lint has version 1.37.1 built from b39dbcd6 on 2021-02-20T11:48:06Z
$ golangci-lint run
foo.go:3:5: `unused` is unused (deadcode)
var unused int
    ^
```

**Comment by ybiquitous** [comment]
@ldez What version of Go?

**Comment by ldez** [comment]
the local version doesn't have any impact because golanci-lint is binary.

```console
$ go version
go version go1.15.8 linux/amd64
$ golangci-lint version
golangci-lint has version 1.37.1 built from b39dbcd6 on 2021-02-20T11:48:06Z
$ golangci-lint run
foo.go:3:5: `unused` is unused (deadcode)
var unused int
    ^
```

```console
$ go version
go version go1.16 linux/amd64
$ golangci-lint version
golangci-lint has version 1.37.1 built from b39dbcd6 on 2021-02-20T11:48:06Z
$ golangci-lint run
foo.go:3:5: `unused` is unused (deadcode)
var unused int
    ^
```

**Comment by ybiquitous** [comment]
Oh, it’s weird. the Docker version seems something wrong... 🤔

**Comment by ldez** [comment]
Sorry but your problem seems offtopic

**Comment by ybiquitous** [comment]
Right. I will create a new issue.
Thank you!

**Comment by chriscasola** [comment]
Is there anything that can be done to help find a solution to this issue? It seems to replicate for us very consistently when running using the golangci-lint-action but never when run on development machines.

[This fix](https://github.com/golangci/golangci-lint/issues/825#issuecomment-666038142) did not solve the problem for us.

[These repro steps](https://github.com/golangci/golangci-lint/issues/825#issuecomment-630004446) did not replicate on our development machines.

Unfortunately the repo we have this issue with is not public.

**Comment by applike-ss** [comment]
For us it was a checksum mismatch due to a re-released dependency under the same tag.

**Comment by dkegel-fastly** [comment]
I just ran into this.  Looks like golangci-lint gets upset if you ask it to lint a subdirectory and it can't find main in a parent directory.

**Comment by chriscasola** [comment]
@dkegel-fastly we have a main package in the root of our project and still encounter this error.

I did notice that we seem to alternate between two different errors on successive runs.

The first is the one mentioned a lot already: `context loading failed: no go files to analyze`.

The second is: `context loading failed: failed to load packages: timed out to load packages: context deadline exceeded`.

We are running with `--timeout 5m` and getting this error well before that limit.

**Comment by tzdybal** [comment]
I was struggling with this issue for some time. As mentioned before, issue was caused by a problem with go.sum (strangely, it was compiling/testing on both my local machine and github workflow).
After cloning my project into "clean machine", when I tried `go test ./...` hash mismatch was detected.

**Comment by gnanderson** [comment]
> I have regularly run `golangci-lint run` from the root of my directory without issue for the last several weeks. Today, after successfully running `golangci-lint run`, I fixed all the reported issues and then ran the command in the same terminal but I got the output:
I'm not sure if this intended behaviour but in 1.39 I experience the same thing unless I specify the "glob" pattern that test/build etc allow
`golanci-lint run ./... `

**Comment by troke12** [comment]
it is only works if repos have `go.mod` ? my repos was literally doesnt have `go.mod` and only some package who located in `vendor folder` here

**Comment by bombsimon** [comment]
> it is only works if repos have `go.mod` ? my repos was literally doesnt have `go.mod` and only some package who located in `vendor folder` here

If you're not using go modules, ensure `GO111MODULES=off` is set. It has previously been set to `auto` before Go 1.16 but is now defaulting to `on`. `auto` means that code in `GOPAT` doesn't require `go.mod` but the rest do. Now Go always requires this unless you explicitly turn it off.

**Comment by troke12** [comment]
> > it is only works if repos have `go.mod` ? my repos was literally doesnt have `go.mod` and only some package who located in `vendor folder` here
> 
> If you're not using go modules, ensure `GO111MODULES=off` is set. It has previously been set to `auto` before Go 1.16 but is now defaulting to `on`. `auto` means that code in `GOPAT` doesn't require `go.mod` but the rest do. Now Go always requires this unless you explicitly turn it off.

yeah, some of modules are really not compatible with go 1.16+, my repos will work only 1.15 above, also how do i turn off that `GO111MODULES=off` in github actions config?

**Comment by bombsimon** [comment]
> > > it is only works if repos have `go.mod` ? my repos was literally doesnt have `go.mod` and only some package who located in `vendor folder` here
> > 
> > 
> > If you're not using go modules, ensure `GO111MODULES=off` is set. It has previously been set to `auto` before Go 1.16 but is now defaulting to `on`. `auto` means that code in `GOPAT` doesn't require `go.mod` but the rest do. Now Go always requires this unless you explicitly turn it off.
> 
> yeah, some of modules are really not compatible with go 1.16+, my repos will work only 1.15 above, also how do i turn off that `GO111MODULES=off` in github actions config?

That's just an environment variable so see [GitHub Actions doc](https://docs.github.com/en/actions/reference/environment-variables).

```yaml
env:
    GO111MODULES: off
```

**Comment by dvonthenen** [comment]
I have a fix for this... it's in PR: https://github.com/vmware-tanzu/tce/pull/918

**Comment by powerman** [comment]
This error happens when `go.mod` is not in sync - i.e. we're added new import into some file but forget to run `go mod tidy` after that. We run linter before tests, so our CI stops with this error message instead of more useful error with some hint about it actual reason (like one shown by `go test ./...`).

So, I propose to improve this error message as a trivial minimal fix: just add a hint to run `go mod tidy` (or, maybe, ask to set `GO111MODULES=off` or run `go mod init` if there is no `go.mod` file at all as an alternative fix) into the message.

**Comment by chriscasola** [comment]
Is there any way I can help come to a solution here? Tips on where to start debugging this? We're only seeing this issue when running in GitHub Actions, but it happens almost constantly there.

**Comment by dkegel-fastly** [comment]
Try only running lint if a regular build succeeds.

**Comment by ansidev** [comment]
same issue: `golangci-lint has version 1.43.0 built from 861262b on 2021-11-02T20:54:42Z`

**Comment by dkegel-fastly** [comment]
See my previous comment.  It's not great to have to do a successful build *before* running lint, but that's how it is...

**Comment by ansidev** [comment]
> See my previous comment. It's not great to have to do a successful build _before_ running lint, but that's how it is...

I've update my comment.

**Comment by dkegel-fastly** [comment]
The problem might be in your githook (or how you are running golangci-lint) rather than in golangci-lint per se... in my quick test just now, golangci-lint run ignored a garbage yaml file.

**Comment by dcherman** [comment]
**This post may only be relevant to people running on Github Actions**

I did a bit of debugging on this and set up SSH on Github Actions if/when linting failed - here's what I found.  By tweaking what linters were running, I eventually ran into an error like the following:

```sh
ERRO Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: open /home/ubuntu/.cache/go-build/a8/a8f00619098a7398f51957837f0043ff4196d4c6cc067214ac77292093507a9f-d: permission denied
```

I did some further debugging, that `/home/ubuntu/.cache/go-build` exists with the wrong ownership (`root:root`) in every workflow that failed.  If I either remove that directory or `sudo chown` it to change the permissions before running my workflow, linting works fine.

Now here's the interesting part - at least in my environment, this folder seems to exist before running *any* actions including checkout, setup go, etc.  The very first thing I did is `ls -lah /home/ubuntu/.cache`, and there it was.  This seems to be a problem with the Github Actions Runner, not with golangci-lint.  I'd be interested to hear if other people observe the same thing when lint fails in Github Actions.

**Comment by jinleileiking** [comment]
3 years later. bug remains.

**Comment by krish7919** [comment]
Same here, `golangci-lint` suddenly breaks on a new project, but was working about a month ago. :/

**Comment by dkegel-fastly** [comment]
krish, does 'go build' work?  If that finds an error, you shouldn't run golangci-lint.
(Also, watch out for go 1.18-related breakage...)

**Comment by krish7919** [comment]
`go build` works.. I am on 1.18.

>  (Also, watch out for go 1.18-related breakage...)

What is this about? Where can I find more information?

**Comment by coder-cdh** [comment]
I tried to clear go.sum ,then executes go mod tidy to resolve the dependency version conflict encountered, and then executes golangci lint run ./...  , It's normal

**Comment by xhd2015** [comment]
> The only advice I could give is to make sure your code is error free, run `go build` or `go test` to make sure that everything works as expected

This answer should be promoted since it clearly addresses that the issue is actually caused by error resides in your project, not that golangci-lint cannot find any go files.

**Comment by WGrape** [comment]
The main reason is that your code failed to compile, maybe you forget to run ```godep save``` or ```go mod download``` in a new container.

I also got this problem with the output when running ```golangci-lint run --timeout=5m``` in gitlab-ci: 

```
level=error msg="Running error: context loading failed: no go files to analyze"
```

When i add the command ```go build``` before ```golangci-lint run --timeout=5m```，and i got this new output :

```
go: cannot find main module, but found Godeps/Godeps.json in xxx/xxx
	to create a module there, run:
	go mod init
```

So, i realized that i need download it's dependencies and compile it successfully.

**Comment by ansidev** [comment]
Bxxxbxeb.cxccbbxxc.ce.x.xbxxcbbbxbxxe.cccxe.bcde.xxe.becb.Becberin ec.ếch. ngồi xbec.Xbox ec.ẽ.hi ẽ.Xbox. xcec.e.xbx.ec.bcyxcec.bxcxex.ec.ceb.xẽ.b.xbbcxbxxcec.cbcbecb.hi ẽ.be.bxbxbxexec.bẽ.bẽb

**Comment by stevekuznetsov** [comment]
Just hit this because I needed to run `go mod tidy`. Would love to see an improvement to the error message. Using v1.49.0

**Comment by dianapicnic** [comment]
In my case the command below helped:

`go mod tidy && go mod vendor `

**Comment by berndschatz** [comment]
i can confirm this solution:

```
[...]
RRO Running error: context loading failed: no go files to analyze 
make: *** [Makefile:46: check-style] Error 5

$ go build
go: updates to go.mod needed; to update it:
	go mod tidy

$ go mod tidy
go: downloading github.com/jhump/protoreflect v1.6.0
go: downloading gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c
go: downloading github.com/xtgo/uuid v0.0.0-20140804021211-a0b114877d4c
go: downloading github.com/google/go-cmp v0.5.6
go: downloading github.com/kr/pretty v0.3.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/sirupsen/logrus v1.8.1
go: downloading github.com/BurntSushi/toml v0.3.1
go: downloading github.com/kr/text v0.2.0
go: downloading github.com/rogpeppe/go-internal v1.8.0
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1

[...]

$ make
Checking for style guide compliance
cd webapp && npm run lint

[...]

plugin built at: .......

```

Error message is not very helpful.

**Comment by nyetwurk** [comment]
We had a race condition in our CI where `go mod tidy` was not run before the cache was created, so `golangci-lint` was being run in an image with no `go/pkg/mod` files. There really should be a better error message for this.

**Comment by dkegel-fastly** [comment]
We just had a problem in CI where, through a long chain of quirks, the source tree was incomplete... and we didn't verify it built before running lint.  Hilarity ensued (with an error like can't run linter goanalysis_metalinter: inspect: failed to load package foo: could not load export data: no export data for...).

Could golangci-lint check somehow that the project had built?  In CI, this could be rather helpful.

**Comment by BedivereZero** [comment]
When obtaining go modules fails, golangci-lint will report the error "no go files to analyze". The error message is inaccurate.

```terminal
$ go mod download
...
verifying devops.example.org/Organization/Project/_git/go-common@v1.0.4: checksum mismatch
	downloaded: h1:Ai9O0FGo9bcG8dkMxjZ2KPPMcrchs3lDItWduc2ht5Q=
	go.sum:     h1:Wto2v4Zv6gZrSppDzaSIJIuwFOMNM77EckvPIoVbsRw=

$ golangci-lint
Running error: context loading failed: no go files to analyze
```

**Comment by ldez** [comment]
`go mod download` != `go mod tidy`

You have to use `go mod tidy`

**Comment by BedivereZero** [comment]
> `go mod download` != `go mod tidy`
> 
> You have to use `go mod tidy`

To reproduce this error, these conditions are required:

1. already run `go mod tidy`, there are go.mod and go.sum in repository's root
2. a go module in go.mod has change，for example the git tag v1.0.4 has been re-created to another git commit
3. there is no go model local cache, like pipeline environment
4. run `golangci-lint run`

**Comment by ldez** [comment]
Can you provide a reproducible example?

**Comment by Integralist** [comment]
I'm seeing this issue currently.

I can run `golangci-lint run --verbose ./cmd/check/...` successfully and the CLI will report back a ton of issues with the code...

But the moment I open that file in Neovim I only see this...

<img width="789" alt="Screenshot 2024-01-11 at 17 56 41" src="https://github.com/golangci/golangci-lint/assets/180050/38fde556-2442-4906-b73e-4aed556c8c24">

**Comment by iewher** [comment]
if golangci-lint gives an error when starting your ci
```bash
Running error: context loading failed: no go files to analyze: running `go mod tidy` may solve the problem
```
add
```bash
    - name: <-- task name -->
      run: go mod tidy && go mod vendor
      working-directory: <-- your-directory -->
```
after run golagci-lint

**Comment by desdic** [comment]
> if golangci-lint gives an error when starting your ci
> 
> Running error: context loading failed: no go files to analyze: running `go mod tidy` may solve the problem
> add
> 
>     - name: <-- task name -->
>       run: go mod tidy && go mod vendor
>       working-directory: <-- your-directory -->
> before run golagci-lint

if you already did go mod vendor and you CI has no internet access then .. yeah guess this is a migration and not fix
